### PR TITLE
Move unexpected MT behavior to troubleshooting sections

### LINKED ISSUE #3006: Move exception behavior from Optimize and Operate MT docs
I’m concerned documenting the exceptional behavior may lead users to conclude that you can run some individual components with multi-tenancy enabled and some without. However, knowing the exceptional behavior could help troubleshoot and debug setups.

We should move this to a troubleshooting section so it's documented but not part of the happy path installation/config.

Internal Slack thread - https://camunda.slack.com/archives/C0555930H7F/p1701894322046489?thread_ts=1701816113.847239&cid=C0555930H7F

### PR DESCRIPTION
**Pull request by akeller**
## Description

Closes #3006.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by akeller** [comment]
Thanks for the discussion today! You'll see some HTML comments in the Operate docs that you can follow up on.

You'll need to do something similar with the Optimize docs, pulling the unexpected or exceptional behavior into a troubleshooting section.

File: docs/self-managed/operate-deployment/operate-configuration.md
Comment (akeller) [inline docs/self-managed/operate-deployment/operate-configuration.md:96]: This should probably be a separate issue unless the documentation already exists and it's easily linkable.

File: docs/self-managed/operate-deployment/operate-configuration.md
Comment (akeller) [inline docs/self-managed/operate-deployment/operate-configuration.md:91]: This is likely something we want to consider for all component docs in MT - routing users back to the concepts page and that really nice, single flag for enabling MT with Helm

**Comment by christinaausley** [comment]
FYI that this will be picked up next week 👍

**Comment by akeller** [comment]
@christinaausley I don't know who will be around next week, so I suggested two reviewers who _might_ be available but are involved with MT. If they are busy or offline, this can wait until I'm back online in Jan.

**Review by koevskinikola — APPROVED** [review]
:+1: @akeller the changes look good (and make sense) to me, but I would defer to @sdorokhova as I am not a maintainer of the Operate part of the docs.

**Comment by christinaausley** [comment]
Still in `draft` here as I'll be making changes this week, but @koevskinikola @sdorokhova I'll let you know when this is ready for final review 👍

**Comment by christinaausley** [comment]
@koevskinikola @sdorokhova @RomanJRW -- interested in your thoughts here.

- Do these rules also apply to your APIs? If so, I'll add a note.
- Would you like to see a separate troubleshooting section? I've added notes to clarify enabling across all components, but not sure if we would prefer this info separated.
- Do you feel this captures the issue at hand? Is there more content you would like to see added? I've tried to be as clear as possible as soon as we refer to multi-tenancy.

**Review by koevskinikola — APPROVED** [review]
@christinaausley the changes look good. 

> Do you feel this captures the issue at hand? Is there more content you would like to see added? I've tried to be as clear as possible as soon as we refer to multi-tenancy.

Having this information across all of the component docs makes sense to me, as we don't know from what C8 application a user may start reading about this feature. I currently can't think of anything else that can be added.

> * Do these rules also apply to your APIs? If so, I'll add a note.
> * Would you like to see a separate troubleshooting section? I've added notes to clarify enabling across all components, but not sure if we would prefer this info separated.

I think a troubleshooting section would be useful for Zeebe as well.

However, due to https://github.com/camunda/zeebe/issues/14285, Zeebe can be used as a standalone application (independent of the Camunda 8 stack) with multi-tenancy, so we have to consider that use case in the troubleshooting section as well.

**Comment by christinaausley** [comment]
In https://github.com/camunda/camunda-docs/pull/3100 we should crosslink from Zeebe docs 👍

**Comment by RomanJRW** [comment]
Going to pass this to @PHWaechtler to comment on the Optimize side

**Comment by PHWaechtler** [comment]
Hi @christinaausley 

> @koevskinikola @sdorokhova @RomanJRW -- interested in your thoughts here.
> 
> * Do these rules also apply to your APIs? If so, I'll add a note.

Sorry, I'm not sure what rules you are referring to, can you clarify? I can't think of anything that would need added to the API docs.

> * Would you like to see a separate troubleshooting section? I've added notes to clarify enabling across all components, but not sure if we would prefer this info separated.

I think the current troubleshooting section you've added works well.

> * Do you feel this captures the issue at hand? Is there more content you would like to see added? I've tried to be as clear as possible as soon as we refer to multi-tenancy.

I think the changes make sense. I did notice that you removed the existing description of what happens when MT is misconfigured in some of the other component's sections when adding the troubleshooting section, but on the Optimize page it is still there (Under `## Default tenant authorizations in Optimize`). I think we dont need to remove it because it gives more context to users to enable them to understand Optimize behaviour, but its also not crucial info so up to you if you'd like to remove this.

**Review by PHWaechtler — APPROVED** [review]

**Comment by christinaausley** [comment]
> > * Do these rules also apply to your APIs? If so, I'll add a note.
> Sorry, I'm not sure what rules you are referring to, can you clarify? I can't think of anything that would need added to the API docs.

For example, we say:

The following configuration is required to enable multi-tenancy in Operate:
| Name                                 | Description                                         | Default value |
| ------------------------------------ | --------------------------------------------------- | ------------- |
| camunda.operate.multiTenancy.enabled | Activates the multi-tenancy feature within Operate. | false         |

:::caution
To ensure seamless integration and functionality, the multi-tenancy feature should also be enabled across all associated components. Find more information, including links to individual component configuration on the [multi-tenancy concepts page](/self-managed/concepts/multi-tenancy.md).
:::

If multi-tenancy is enabled across components, users are allowed to view any data from tenants for which they have authorizations configured in Identity.

If multi-tenancy is disabled in Operate, all users are allowed to view data from the `<default>` tenant only and no data from other tenants.

**The same rules apply to the [Operate API](../../apis-tools/operate-api/overview.md#multi-tenancy).**

> > * Do you feel this captures the issue at hand? Is there more content you would like to see added? I've tried to be as clear as possible as soon as we refer to multi-tenancy.
> 
> I think the changes make sense. I did notice that you removed the existing description of what happens when MT is misconfigured in some of the other component's sections when adding the troubleshooting section, but on the Optimize page it is still there (Under `## Default tenant authorizations in Optimize`). I think we dont need to remove it because it gives more context to users to enable them to understand Optimize behaviour, but its also not crucial info so up to you if you'd like to remove this.

**Comment by PHWaechtler** [comment]
@christinaausley 
> For example, we say:
> 
> The following configuration is required to enable multi-tenancy in Operate:
> 
> Name	Description	Default value
> camunda.operate.multiTenancy.enabled	Activates the multi-tenancy feature within Operate.	false
> If multi-tenancy is disabled in Operate, all users are allowed to view data from the `<default>` tenant only and no data from other tenants.
> 
> **The same rules apply to the [Operate API](../../apis-tools/operate-api/overview.md#multi-tenancy).**

Ah gotcha. The API does not take into account user authorizations or MT configurations directly, so for example for report result export, it evaluates the report as is. The data returned just depends on the tenants the user selected when building the report - which in turn depends on the MT configuration and each user's tenant auths. If that changes in the future we'll adjust the docs then though.

> The content is still "here", but it is moved around a bit. Or, is something clearly removed and missing here? I may just be missing it.

I think I just overlooked that then all good. :)

**Review by akeller — COMMENTED** [review]
Love the improvements here! But because I'm the author of the PR, I can't approve 😆 

Thank you for the great work and collaboration here! The cross-linking and inclusion of the API functionality too is excellent for the docs journey.

**Review by christinaausley — APPROVED** [review]

**Pull request merged**
